vector-db-proxy/src/vector_databases/models.rs

```diff
@@ -350,24 +363,28 @@ pub struct CollectionCreate {
     pub namespace: Option<String>,
     pub distance: Distance,
     pub vector_name: Option<String>,
-    pub region: Option<Region>,
-    pub cloud: Option<Cloud>,
+    pub region: Option<String>,
+    pub cloud: Option<String>,
+    pub index_name: Option<String>,
 }
 impl CollectionCreate {
     pub fn new(
         collection_name: String,
         dimensions: usize,
         distance: Distance,
-        region: Region,
+        region: String,
+        cloud: String,
+        index_name: String,
     ) -> Self {
         Self {
             collection_name: collection_name.clone(),
             dimensions,
             distance,
             namespace: None,
             vector_name: None,
-            cloud: Some(Cloud::AWS),
+            cloud: Some(cloud),
             region: Some(region),
+            index_name: Some(index_name),
```

Comment:
This is exactly what i did NOT want to do. i don't want a design where we have vendor specific language. What we want is a generic design that each vendor adopts in their own implementation of the trait